electron/localDB.ts

```diff
@@ -0,0 +1,175 @@
+import Database from 'better-sqlite3';
+import { app } from 'electron';
+import { existsSync, unlinkSync } from 'fs';
+import { createRequire } from 'node:module';
+import path from 'node:path';
+import { pathToFileURL } from 'node:url';
+import { log } from './log';
+
+// 2023-08-21 the following options are a fix for using rollup (within vite) with better-sqlite3
+const requireMe = createRequire(pathToFileURL(__filename).href);
+const addon = requireMe(
+    path.resolve('./node_modules/better-sqlite3/build/Release/better_sqlite3.node').replace(/(\.node)?$/, '.node'),
+);
+
+const DB_PATH = (MODE) => path.join(app.getPath('userData'), `bahis_${MODE}.db`);
+
+export const createLocalDatabase = (MODE) => {
+    log.info(`CREATE clean local database at ${DB_PATH(MODE)}`);
+    const db = new Database(DB_PATH(MODE), { nativeBinding: addon });
+
+    log.info('Running initialisation');
+    try {
+        initialiseDBTables(db);
+        log.info('Running initialisation SUCCESS');
+    } catch (error) {
+        log.error('Running initialisation FAILED');
+        log.error(error);
+    }
+
+    log.info('CREATE clean local database FINSIHED');
+    return db;
+};
+
+export const createOrReadLocalDatabase = (MODE) => {
+    log.info(`Checking if local database exists at ${DB_PATH(MODE)}`);
+    let db;
+    if (!existsSync(DB_PATH(MODE))) {
+        db = createLocalDatabase(MODE);
+    } else {
+        log.info(`Using existing local database at ${DB_PATH(MODE)}`);
+        db = new Database(DB_PATH(MODE), { nativeBinding: addon });
+    }
+
+    return db;
+};
+
+export const createUserInLocalDatabase = async (data, userData, db) => {
+    const insertStmt = db.prepare(`INSERT INTO user (username, password, name, role, upazila) VALUES (?, ?, ?, ?, ?)`);
```

Comment:
what is the user table name? maybe `users`

Reply:
Ah, okay, sorry, it looks like I did migrate everything over to the new simpler BAHIS 3 `user` table but must have left one or two references to the `users2` table. Please repoint all `users2` references to `user` (and check it works :rofl: )